src/components/templates/PartyMap/components/CanvasMap/CanvasMap.tsx

```diff
@@ -0,0 +1,291 @@
+            roomImageBackgroundTexture.baseTexture.height * 0.66
+          );
+          text.anchor.set(0.5, 0.5);
+
+          roomSprite.interactive = true;
+          roomSprite.buttonMode = true;
+          roomSprite.on("click", (event: any) => {
+            setSelectedRoom(room);
+            setIsRoomModalOpen(true);
+          });
+
+          roomSprite.anchor.set(0.5, 0.5);
+
+          roomSprite.addChild(roomImageBackgroundSprite);
+          roomSprite.addChild(text);
+
+          roomSprite.on("mouseover", () => {
+            roomSprite.filters = [new Filters.OutlineFilter(2, 0xffffff)];
+          });
+
+          roomSprite.on("mouseout", () => {
+            roomSprite.filters = [];
+          });
+
+          roomSprites.push([
+            roomSprite,
+            () => {
+              setSelectedRoom(room);
+              setIsRoomModalOpen(true);
+            },
+          ]);
+
+          newApp.stage.addChild(roomSprite);
+        };
+      });
+    };
+
+    // const rectangle = PIXI.Sprite.from(PIXI.Texture.WHITE);
+    // rectangle.width = 300;
+    // rectangle.height = 200;
+    // rectangle.x = 60;
+    // rectangle.y = 60;
+    // rectangle.tint = 0xff0000;
+    // roomSprites.push([rectangle, () => {}]);
+    // newApp.stage.addChild(rectangle);
+
+    const player = createPlayer({
+      id: user.id,
+      userProfile: user,
+      onCreated: (child) => newApp.stage.addChild(child),
+      isMe: true,
+    });
+
+    const checkCollisionPlace = (
+      a: PIXI.Rectangle,
+      b: PIXI.Rectangle
+    ): boolean => {
+      return (
+        a.x + a.width > b.x &&
+        a.x < b.x + b.width &&
+        a.y + a.height > b.y &&
+        a.y < b.height + b.y
+      );
+    };
+
+    const makeOnIntersectCallback = (
+      player: PIXI.Sprite,
+      venue: PIXI.Sprite,
+      callback: () => void
+    ) => {
+      const playerBox = player.getBounds();
+
+      const venueBox = venue.getBounds();
+      const collision = checkCollisionPlace(playerBox, venueBox);
+
+      if (collision) {
+        callback();
+
+        player.position.set(5, 5);
+      }
+    };
+
+    const startGame = () => {
+      on("KeyW", () => (player.y -= 3));
+      on("KeyS", () => (player.y += 3));
+      on("KeyD", () => (player.x += 3));
+      on("KeyA", () => (player.x -= 3));
+
+      roomSprites.forEach(([room, callback]) =>
+        makeOnIntersectCallback(player, room, callback)
+      );
+
+      charm.update();
+
+      requestAnimationFrame(startGame);
+    };
+
+    startGame();
+
+    divElement.appendChild(newApp.view);
+    appRef.current = {
+      app: newApp,
+      charm,
+    };
+
+    // const updatePositionIntervalId = setInterval(() => {
+    //   firestore
+    //     .collection("venues")
+    //     .doc(venue.name)
+    //     .collection("onlinePlayers")
+    //     .doc(user.id)
+    //     .set({
+    //       x: player.x,
+    //       y: player.y,
+    //     });
+    // }, 200);
```

Comment:
You don't need to have this in an interval, you could just have it in a `useEffect` or similar, with the player's position in the `deps`.